### PR TITLE
Check for nil decoder before use

### PR DESCRIPTION
**Pull request by jlk**
simple add of if test before using decoder.

### CODE REVIEW

File: pkg/utils/yaml.go [hunk 1]
```diff
@@ -102,6 +102,11 @@ func ScanIacDocumentsFromYaml(scanner *bufio.Scanner, byteArray []byte, filePath
 
 	dec := yaml.NewDecoder(bytes.NewReader(byteArray))
 
+	if dec == nil {
```
Comment (patilpankaj212) [inline pkg/utils/yaml.go:105]: Though the nil check is good, looking at the yaml.NewDecoder method, it will either panic or return a non nil decoder.

**Review by patilpankaj212 — COMMENTED** [review]
added review.

**Comment by dev-gaur** [comment]
I think the CI will pass here even if you don't include the changes in go.mod and go.sum . I understand that running `make validate` locally must have required a `go mod vendor` first.

File: pkg/utils/yaml.go [hunk 2]
```diff
@@ -102,6 +102,11 @@ func ScanIacDocumentsFromYaml(scanner *bufio.Scanner, byteArray []byte, filePath
...
+		err := fmt.Errorf("error cretaing yaml decoder in ScanIacDocumentsFromYaml()")
```
Comment (dev-gaur) [inline pkg/utils/yaml.go:106]: hey @jlk , I think the user wouldn't care to know the function name here. so we can keep the error message generic and simple. also, creating is mispelled

Comment (dev-gaur) [reply pkg/utils/yaml.go:106]: requsting this change

**Review by dev-gaur — CHANGES_REQUESTED** [review]

**Comment by jlk** [comment]
Sorry for ignoring this. Upon looking closer, the issue was in the library we're using, so will open a PR with them. Closing this PR.

**Comment by dev-gaur** [comment]
#611

**Pull request closed without merging**